src/mdlComponentHandler.js

```diff
@@ -389,7 +374,15 @@ componentHandler = (function() {
      * @param  {!Node} node the node to be downgraded
      */
     var downgradeNode = function(node) {
-      deconstructComponentInternal(findCreatedComponentByNodeInternal(node));
+      var found = [];
+      createdComponents_.forEach(function(item) {
+        if (item.element_ === node) {
+          found.push(item);
```

Comment:
Yup, it isn't working unless we collect all the component's first then downgrade.

![selection_002](https://cloud.githubusercontent.com/assets/868301/10569220/b19ebf84-75f0-11e5-9c4f-f75b2cd8a6db.png)

with: 

```javascript
  function downgradeNodesInternal(nodes) {
    /**
     * Auxiliary function to downgrade a single node.
     * @param  {!Node} node the node to be downgraded
     */
    var downgradeNode = function(node) {
      createdComponents_.forEach(function(item) {
        if (item.element_ === node) {
          deconstructComponentInternal(item);
        }
      });
    };
    if (nodes instanceof Array || nodes instanceof NodeList) {
      for (var n = 0; n < nodes.length; n++) {
        downgradeNode(nodes[n]);
      }
    } else if (nodes instanceof Node) {
      downgradeNode(nodes);
    } else {
      throw new Error('Invalid argument provided to downgrade MDL nodes.');
    }
  }
```